app/uk/gov/hmrc/helptosavestub/controllers/SquidController.scala

```diff
@@ -0,0 +1,304 @@
+        case _ => true
+      })
+
+    //        val noSpacesPostcode = postcode.filter{_ != ' '}
+    //        val postcodeRegex =     ("""^((GIR)(0AA)|([A-PR-UWYZ]([0-9]{1,2}|([A-HK-Y][0-9]|[A-HK-Y][0-9]([0-9]|[ABEHMNPRV-Y]))|""" +
+    //          """[0-9][A-HJKS-UW]))([0-9][ABD-HJLNP-UW-Z]{2})|(([A-Z]{1,4})(1ZZ))|((BFPO)([0-9]{1,4})))$""").r
+    //        !postcodeRegex.pattern.matcher(noSpacesPostcode).matches()
+  }
+
+  private def unparsableLocalDate(str: String) = {
+    !"""^\d{8}$""".r.pattern.matcher(str).matches()
+  }
+
+  private def yearFromDate(date: String): Try[Int] = {
+    if (unparsableLocalDate(date)) {
+      Failure(new IllegalArgumentException())
+    } else {
+      Try(date.substring(0, 4).toInt)
+    }
+  }
+
+  private def monthFromDate(date: String): Try[Int] = {
+    if (unparsableLocalDate(date)) {
+      Failure(new IllegalArgumentException())
+    } else {
+      Try(date.substring(4, 6).toInt)
+    }
+  }
+
+  private def dayFromDate(date: String): Try[Int] = {
+    if (unparsableLocalDate(date)) {
+      Failure(new IllegalArgumentException())
+    } else {
+      Try(date.substring(6, 8).toInt)
+    }
+  }
+
+  private def invalidDay(date: String): Boolean = {
+    val dayNumber: Try[Int] = dayFromDate(date)
+    val monthNumber = monthFromDate(date)
+    val yearNumber = yearFromDate(date)
+
+    if (dayNumber.isSuccess && monthNumber.isSuccess && yearNumber.isSuccess) {
+      val day = dayNumber.getOrElse(0)
+      val month = monthNumber.getOrElse(0)
+      val year = yearNumber.getOrElse(0)
+      val firstDayOfMonthDate = java.time.LocalDate.of(year, month, 1)
+      val lastDayOfMonthDate = firstDayOfMonthDate.`with`(TemporalAdjusters.lastDayOfMonth())
+      val lastDay = lastDayOfMonthDate
+      day < 1 || day > lastDay.getDayOfMonth
+    } else {
+      true
+    }
+  }
+
+  private def invalidCentury(date: String): Boolean = {
+    val yearNumber = yearFromDate(date)
+
+    if (yearNumber.isSuccess) {
+      val year = yearNumber.getOrElse(0)
+      year < 1800 || year > 2099
+    } else {
+      true
+    }
+  }
+
+  private def invalidNino(nino: String): Boolean = {
+    !ninoRegex.pattern.matcher(nino).matches()
+  }
+
+  private def invalidCommunicationPreference(pref: String): Boolean = {
+    pref != "00" && pref != "02"
+  }
+
+  private def emailRequired(pref: String, emailAddress: Option[String]) = {
+    pref == "02" && emailAddress.isEmpty
+  }
+
+  //Helper method
+  private def mt(code: String, mk0: String, mk1: String): (String, String, String) = (code, messagesApi(mk0), messagesApi(mk1))
+
+  private def validateCreateAccount(json: JsValue): Either[(String, String, String), AccountCommand] = {
+    //Convert incoming json to a case class
+    val parseResult = Json.fromJson[AccountCommand]((json \ "createAccount").get)
+
+    logger.info(json.toString())
+
+    parseResult match {
+      case JsSuccess(createAccount, _) =>
+        if (createAccount.forename.startsWith(" ")) {
```

Comment:
I believe not - once again I was trying to produce the errors that NS&I will produce. The built in validation will fail (on dates in particular) because it will not validate  stuff that is wrong, rather than allowing me to produce the NS&I error code for the particular kind of failure.